mlflow/server/js/src/Actions.js

```diff
@@ -52,12 +52,36 @@ export const getRunApi = (runUuid, id = getUUID()) => {
   };
 };
 
+export const DELETE_RUN_API = 'DELETE_RUN_API';
+export const deleteRunApi = (runUuid, id = getUUID()) => {
+  return (dispatch) => {
+    const deleteResponse = dispatch({
+      type: DELETE_RUN_API,
+      payload: wrapDeferred(MlflowService.deleteRun, { run_id: runUuid }),
+      meta: { id: getUUID() },
+    });
+    return deleteResponse.then(dispatch(getRunApi(runUuid, id)));
```

Comment:
Ah cool, neat to see this example of how we can chain together promises with redux-promise-middleware & redux-thunk :). I see what you meant earlier about whether `dispatch` will return a promise (i.e. if `deleteResponse` will be a promise) - I don't think `redux-promise-middleware` guarantees this in any API docs, but experimentally it does seem to be the case. [This example](https://github.com/pburtchaell/redux-promise-middleware/blob/master/docs/guides/rejected-promises.md#catching-errors-locally) also suggests that `dispatch` will return a Promise.

Reply:
Do you know what best practices are for error-handling on the `getRunApi` action? [This example](https://github.com/pburtchaell/redux-promise-middleware/blob/master/docs/guides/rejected-promises.md#catching-errors-locally) seems to suggest that we should have error-handling logic in the action creator itself (i.e. not [our existing pattern](https://github.com/mlflow/mlflow/blob/master/mlflow/server/js/src/components/modals/RenameRunModal.js#L64), which relies on e.g. `dispatch(setTagApi())` returning a promise & calls try/catch on that promise in individual components). However I think the right error-handling behavior is often specific to each component, so not sure it makes sense to always handle errors in action creators